src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVCollapser.java

```diff
@@ -374,22 +414,203 @@ private static int compareCopyNumberForCollapsing(final Integer first, final Int
+            copyNumber = collapseSampleCopyNumber(genotypes, expectedCopyNumber);
+            builder.attribute(GATKSVVCFConstants.COPY_NUMBER_FORMAT, copyNumber);
+        } else {
+            copyNumber = null;
+        }
+        final List<Allele> collapsedAlleles = collapseSampleGenotypeAlleles(genotypes, expectedCopyNumber, copyNumber, altAlleles, refAllele);
+        // Replace ref alleles with the possibly-new one
+        final List<Allele> collapsedAllelesNewRef = collapsedAlleles.stream()
+                .map(a -> (a != null && a.isReference()) ? refAllele : a)
+                .collect(Collectors.toList());
+        builder.alleles(collapsedAllelesNewRef);
         return builder.make();
     }
 
+    /**
+     * Generates genotype alleles, i.e. for the GT field, for CNVs (DEL and/or DUP). Genotypes that cannot be determined
+     * unambiguously (e.g. at diploid DUP sites) result in {@link Allele#NO_CALL} alleles.
+     * @param siteAltAlleles  unique alt alleles for the sample at the site
+     * @param refAllele  reference allele
+     * @param expectedCopyNumber  expected copy number (i.e. ploidy)
+     * @param copyNumber  copy number state
+     * @return  alleles for the sample at the site
+     * @throws {@link IllegalArgumentException} if the alt allele(s) are not CNV(s)
+     */
+    public static List<Allele> getCNVGenotypeAllelesFromCopyNumber(final List<Allele> siteAltAlleles,
+                                                                   final Allele refAllele,
+                                                                   final int expectedCopyNumber,
+                                                                   final Integer copyNumber) {
+        Utils.nonNull(siteAltAlleles);
+        Utils.nonNull(refAllele);
+        Utils.validateArg(siteAltAlleles.size() <= 2, "No support for variants with over 2 alleles");
+        if (copyNumber == null) {
+            // Had conflicting genotypes
+            return Collections.nCopies(expectedCopyNumber, Allele.NO_CALL);
+        }
+        Utils.validateArg(copyNumber >= 0, "Invalid negative copy number: " + copyNumber);
+        Utils.validateArg(expectedCopyNumber >= 0, "Invalid expected copy number: " + expectedCopyNumber);
+        if (siteAltAlleles.isEmpty()) {
+            return Collections.nCopies(expectedCopyNumber, refAllele);
+        } else if (siteAltAlleles.size() == 2) {
+            if (siteAltAlleles.contains(Allele.SV_SIMPLE_DUP) && siteAltAlleles.contains(Allele.SV_SIMPLE_DEL)) {
+                return getMultiallelicCNVAllelesFromCopyNumber(refAllele, expectedCopyNumber, copyNumber);
+            } else {
+                final String messageAlleles = String.join(", ",
+                        siteAltAlleles.stream().map(Allele::getDisplayString).collect(Collectors.toList()));
+                throw new IllegalArgumentException("Unsupported CNV alt alleles: " + messageAlleles);
```

Comment:
It might be good to throw in a unit test that hits this (unless you already did and I missed it).